### DISCUSSION TITLE
Couldn't run pyscenic aucell

### DISCUSSION THREAD

**Discussion by baijiexu** [aertslab/SCENIC#345] [Python]
Hi, 
When I run `pyscenic aucell`, I get the error here.

`pyscenic.cli.pyscenic - ERROR - A gene signature must have at least one gene.`

I really don't know how to do next. I could run successfully for `pyscenic grn` and `pyscenic ctx`

However, I was stuck at `pyscenic aucell` this step. Could you help me figure it out?

Here is all the information:
````
pyscenic aucell simcre_ctrl.loom simcre_ctrl_reg.csv --output simcre_ctrl_SCENIC.loom --num_workers 20

2022-09-30 23:16:34,077 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.

2022-09-30 23:16:34,808 - pyscenic.cli.pyscenic - INFO - Loading gene signatures.
Create regulons from a dataframe of enriched features.
Additional columns saved: []

2022-09-30 23:16:42,458 - pyscenic.cli.pyscenic - ERROR - A gene signature must have at least one gene.
```
`

I appreciate, thank you so much!

BJ

**Comment by nitinnarwade** [comment]
Hi everyone,

I ran into the same error and managed to fix it. As nobody has replied to this issue yet, I’m sharing my solution in case it helps someone else in the future.

To fix the error, open the `transform.py` file and find this line (**line number 520 in pyscenic v0.12.1**):

`df[COLUMN_NAME_TYPE] = df.apply(get_type, axis=1)`

Right after that line, add this:

`df = df[df["TargetGenes"].astype(bool)]`



**Explanation:**
The error occurs because some motif scan results are not properly filtered and include entries with zero target genes. The added line filters out any rows in the DataFrame where the "TargetGenes" list is empty, ensuring only valid entries are processed.

Regards,
Nitin N.
